contracts/VRTVault/VRTVault.sol

```diff
@@ -0,0 +1,237 @@
+        uint256 accruedInterest = getAccruedInterest(userAddress);
+        if(accruedInterest > 0){
+            UserInfo storage user = userInfo[userAddress];
+            user.totalInterestAmount = user.totalInterestAmount.add(accruedInterest);
+            uint256 vrtBalance = vrt.balanceOf(address(this));
+            require(vrtBalance >= accruedInterest, "Failed to transfer VRT, Insufficient VRT in Vault.");
+            emit Claim(userAddress, accruedInterest);
+            user.accrualStartBlockNumber = getBlockNumber();
+            vrt.safeTransfer(user.userAddress, accruedInterest);
+        }
+    }
+
+    /**
+     * @notice withdraw accruedInterest and totalPrincipalAmount of the user's VRTDeposit in the Vault
+     */
+    function withdraw() external nonReentrant isInitialized userHasPosition(msg.sender) {
+        address userAddress = msg.sender;
+        uint256 accruedInterest = getAccruedInterest(userAddress);
+
+        UserInfo storage user = userInfo[userAddress];
+
+        if(accruedInterest > 0){
+            user.totalInterestAmount = user.totalInterestAmount.add(accruedInterest);
+        }
+
+        uint256 totalPrincipalAmount = user.totalPrincipalAmount;
+        uint256 vrtForWithdrawal = accruedInterest.add(totalPrincipalAmount);
+
+        uint256 vrtBalance = vrt.balanceOf(address(this));
+        require(vrtBalance >= vrtForWithdrawal, "Failed to transfer VRT, Insufficient VRT in Vault.");
+
+        emit Withdraw(userAddress, vrtForWithdrawal, totalPrincipalAmount, accruedInterest);
+        vrt.safeTransfer(user.userAddress, vrtForWithdrawal);
+    }
+
+    /**
+     * @notice withdraw BEP20 tokens from the contract to a recipient address.
+     * @param tokenAddress address of the BEP20 token
+     * @param receiver recipient of the BEP20 token
+     * @param amount tokenAmount
+     */
+    function withdrawBep20(address tokenAddress, address receiver, uint256 amount) onlyAdmin isInitialized nonZeroAddress(tokenAddress) nonZeroAddress(receiver) public {
+        require(amount > 0 , "amount is invalid");
+        IBEP20 token = IBEP20(tokenAddress);
+        require(amount <= token.balanceOf(address(this)), "Insufficient amount in Vault");
+        emit WithdrawToken(tokenAddress, receiver, amount);
+        token.safeTransfer(receiver, amount);
+    }
+
+    /**
+     * @dev Returns the address of the current admin
+     */
+    function getAdmin() public view returns (address) {
+        return admin;
+    }
+
+    /**
+     * @dev Burn the current admin
+     */
+    function burnAdmin() public onlyAdmin {
+        emit AdminTransfered(admin, address(0));
+        admin = address(0);
+    }
+
+    /**
+     * @dev Set the current admin to new address
+     */
+    function setNewAdmin(address newAdmin) public onlyAdmin {
+        require(newAdmin != address(0), "new owner is the zero address");
+        emit AdminTransfered(admin, newAdmin);
+        admin = newAdmin;
+    }
```

Comment:
We can use proxy functions to get and set admins, do we really need them in the proxy? `setNewAdmin` removes two-step admin change, and emits an event different from the one that proxy emits.

Reply:
@kkirka yeah unused.  these were the functions added from XVSVault / VAIVault contracts
i will remove these.